mat-process-api.Tests/V1/Controllers/ProcessDataControllerTests.cs

```diff
@@ -73,12 +69,114 @@ public void given_a_processRef_when_getprocessdata_method_is_called_it_then_call
+
+            //act
+            IActionResult controllerResponse = _processDataController.GetProcessData(requestObject);
+            var result = (ObjectResult)controllerResponse;
+            var actualStatusCode = result.StatusCode;
+
+            //Assert
+            Assert.NotNull(controllerResponse);
+            Assert.NotNull(result);
+            Assert.AreEqual(expectedStatusCode, actualStatusCode);
+        }
+
+        [Test]
+        public void given_GetProcessData_controller_method_call_when_request_validation_fails_then_the_controller_returns_correct_badRequest_response()
+        {
+            //arrange
+            int errorCount = faker.Random.Int(1, 10); //simulate from 1 to 10 validation errors (triangulation).
+            var validationErrorList = new List<ValidationFailure>(); //this list will be used as constructor argument for 'ValidationResult'.
+            for (int i = errorCount; i > 0; i--) { validationErrorList.Add(new ValidationFailure(faker.Random.Word(), faker.Random.Word())); } //generate from 1 to 10 fake validation errors. Single line for-loop so that it wouldn't distract from what's key in this test.
+
+            var fakeValidationResult = new FV.ValidationResult(validationErrorList); //Need to create ValidationResult so that I could setup Validator mock to return it upon '.Validate()' call. Also this is the only place where it's possible to manipulate the validation result - You can only make the validation result invalid by inserting a list of validation errors as a parameter through a constructor. The boolean '.IsValid' comes from expression 'IsValid => Errors.Count == 0;', so it can't be set manually.
+            _mockValidatorGet.Setup(v => v.Validate(It.IsAny<GetProcessDataRequest>())).Returns(fakeValidationResult);
+
+            string processRef = faker.Random.String();
+
+            GetProcessDataRequest request = new GetProcessDataRequest() { processRef = processRef };
+
+            //act
+            var controllerResponse = _processDataController.GetProcessData(request);
+            var result = (ObjectResult)controllerResponse;
+            var resultContents = (IList<ValidationFailure>)result.Value;
+
+            //assert
+            Assert.NotNull(controllerResponse);
+            Assert.IsInstanceOf<BadRequestObjectResult>(result); // should be bad request, since validation fails.
+            Assert.NotNull(result);
+            Assert.IsInstanceOf<IList<ValidationFailure>>(resultContents);
+            Assert.NotNull(resultContents);
+
+            Assert.AreEqual(errorCount, resultContents.Count); // there should be 1 validation failure coming from the invalid guid
+            Assert.AreEqual(400, result.StatusCode); // expecting bad request result
+        }
+
+        [Test]
+        public void test_that_controller_returns_correct_message_and_status_code_when_no_document_was_found()
```

Comment:
More specific test name would be good. Like:
`given_DocumentNotFound_exception_is_thrown_when_GetProcessData_controller_method_returns_404_status_code_with_correct_error_message`